Unable to build with `make`

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/appcelerator/amp/blob/master/project/CONTRIBUTING.md#reporting-other-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- AMP Support Forums - https://amp.mobilize.io
- IRC - irc.freenode.net #amp-cloud channel
- Post a question on StackOverflow, using the Docker tag

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

Hi guys, I was unable to build the code on my Mac with `make` command after cloning and checking out master.

```
git clone https://github.com/appcelerator/amp
Cloning into 'amp'...
remote: Counting objects: 29944, done.
remote: Compressing objects: 100% (102/102), done.
ampeiving objects:   3% (1138/29944), 628.00 KiB | 580.00 KiB/s     
remote: Total 29944 (delta 70), reused 19 (delta 19), pack-reused 29822
Receiving objects: 100% (29944/29944), 105.54 MiB | 3.81 MiB/s, done.
Resolving deltas: 100% (13558/13558), done.
mini:appcelerator alex$ cd amp
mini:amp alex$ make
version: v0.6.0-dev (build: 95583605)
Unable to find image 'appcelerator/gotools2:1.2.0' locally
1.2.0: Pulling from appcelerator/gotools2
c952bb7239f0: Pull complete 
df674db1f5c5: Pull complete 
a97590af2dda: Pull complete 
2727f2714cc4: Pull complete 
56e4f3b398ed: Pull complete 
6d3c63919bcc: Pull complete 
7290b31bf5a4: Pull complete 
25e9742e8526: Pull complete 
b20bb53e8fa4: Pull complete 
e0deb870d58d: Pull complete 
0fd70c64be66: Pull complete 
Digest: sha256:6f0680acad89e0e447953893abba103eee69c86868eac7b8d3944e4470c3d4df
Status: Downloaded newer image for appcelerator/gotools2:1.2.0
hack/build: line 5: $1: unbound variable
make: *** [build-ampadm] Error 1
mini:amp alex$
```

Here's some debugging info:


```
mini:amp alex$ uname -a
Darwin mini.local 16.3.0 Darwin Kernel Version 16.3.0: Thu Nov 17 20:23:58 PST 2016; root:xnu-3789.31.2~1/RELEASE_X86_64 x86_64
mini:amp alex$ docker version
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Wed Jan 18 16:20:26 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Wed Jan 18 16:20:26 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

--- a/.gitignore
+++ b/.gitignore
@@ -6,6 +6,9 @@
 /amp-function-listener
 /amp-function-worker
 /amp-cluster
+/adm-agent
+/adm-server
+/ampadm
 /dist
 /*.exe
 coverage.out

--- a/Makefile
+++ b/Makefile
@@ -186,7 +186,7 @@ build-adm-server: proto
 build-adm-agent: proto
 	@hack/build $(CLUSTERAGENT)
 build-ampadm: proto
-	@hack/build $(CLUSTERADM)
+	@hack/build $(AMPADM)
 build-fn-listener: proto
 	@hack/build $(FUNCTION_LISTENER)
 build-fn-worker: proto

--- a/hack/proto.go
+++ b/hack/proto.go
@@ -48,7 +48,7 @@ func main() {
 	flag.Parse()
 
 	// run protoc on command line args if provided
-	if len(os.Args) > 0 {
+	if flag.NArg() > 0 {
 		for _, p := range os.Args[1:] {
 			// filter out options
 			if !strings.HasPrefix(p, "-") {

